### PR TITLE
docs(new): Add TSDoc to Coverage class

### PR DESCRIPTION
**Pull request by hashseed**

### CODE REVIEW

File: src/common/Coverage.ts
Comment (hashseed) [inline src/common/Coverage.ts]: This doesn't seem to be the right syntax. How do we document a bag of options?

File: src/common/Coverage.ts
Comment (jackfranklin) [inline src/common/Coverage.ts]: there's no need to document these here - the comments you've added below will do the trick.

What you do need is to mark this as `@public` and also `export` the interface too.

File: src/common/Coverage.ts
Comment (jackfranklin) [inline src/common/Coverage.ts:77]: I think the `@remark` should go before the example so the summary (the stuff pre-remark) is nice and concise.

File: src/common/Coverage.ts
Comment (jackfranklin) [reply src/common/Coverage.ts]: The best fix is to extract an interface for these options so you can document them properly. Else TSDoc doesn't handle it very well :(

Comment (hashseed) [reply src/common/Coverage.ts]: Done.

File: src/common/Coverage.ts
Comment (hashseed) [reply src/common/Coverage.ts]: Done.

File: src/common/Coverage.ts
Comment (hashseed) [reply src/common/Coverage.ts:77]: Done.

**Comment by hashseed** [comment]
Please re-review :)

**Comment by jackfranklin** [comment]
@hashseed LGTM! I think after you merged in `main` you need to re-generate the docs and then CI will be happy.

**Review by jackfranklin — APPROVED** [review]

**Pull request merged**
